### ISSUE TITLE
Walmart Press & Hold cpatcha

### ISSUE THREAD

**Issue by senzacionale** [berstend/puppeteer-extra#712]
Is it possible to support also Walmart Press & Hold captcha?

![image](https://user-images.githubusercontent.com/6663030/190483732-b1d60950-a233-4c19-b3c0-c1b9c14ca65f.png)

Thank you

**Comment by jinmvp** [comment]
You can try switch to recaptcha V2 and solve it.

**Comment by senzacionale** [comment]
Hi. Thank you. How do you mean switch to recaptcha v2?

**Comment by jinmvp** [comment]
> Hi. Thank you. How do you mean switch to recaptcha v2?

You can try with this link: [Here](https://www.walmart.com/blocked?url=L2FjY291bnQvbG9naW4/dmlkPW9hb2gmcmVmPWRvbWFpbg==)
There is a try a different method button there.

**Comment by senzacionale** [comment]
I try to parse a webpage that has Walmart press & hold capctha. So you mean to use logic to resolve this captcha from recaptcha v2?

**Comment by jinmvp** [comment]
Which site you wanna try? Give me the link.

**Comment by senzacionale** [comment]
Thank you for this idea. But unfortunatelly Press and Hold captcha can not be resolved with Recaptcha v2. 

> Which site you wanna try? If its walmart, i already gave u the solution, just read my words carefully.

**Comment by senzacionale** [comment]
No I want to resolve this Press and Hold captcha. I contact also 2captcha.com support if they have solution. 

Answer:
 
'This is a PerimeterX protection.
Currently we don't have any method to bypass that.'

It looks like this captcha can not be resolved. 


> This looks very similar to the "press and hold" feature on the ssense.com that I can't seem to figure out :sob: . 
> 
> 
> 
> > You can try switch to recaptcha V2 and solve it.
> 
> >> How do you mean switch to recaptcha v2?
> 
> 
> 
> @jinmvp I think @senzacionale is asking how to switch to recaptcha v2. Is that a separate configuration option? I have `provider.id` option set to `2captcha` provider, so v2 would be used automatically when running `await page.solveRecaptchas()`, is that right?

**Comment by markcellus** [comment]
Yeah, I don't think the `puppeteer-extra-plugin-recaptcha` package can solve this one. Here's the similar one from SSENSE.com. It has the captcha in an iframe inside of a _closed_ shadow DOM :disappointed: 

![image](https://user-images.githubusercontent.com/1446916/193680571-0bc8a4c0-508e-4470-bf3d-8d74d8b5d378.png)

**Comment by markcellus** [comment]
@senzacionale I was able to get around it by moving the mouse over the "Press and hold" button and holding it down for a few seconds. Here is my code, but you'll need to update the mouse move coordinates to exactly where the button appears on the page.

```js
await page.mouse.move(200, 450);
await page.mouse.down();
// hold down for six seconds
await new Promise((resolve) => setTimeout(resolve, 6000));
```

Hope this helps!

**Comment by senzacionale** [comment]
@markcellus thank you. I will try and let you know.

**Comment by jinmvp** [comment]
> This looks very similar to the "press and hold" feature on the ssense.com that I can't seem to figure out 😭 .
> 
> > You can try switch to recaptcha V2 and solve it.
> > > How do you mean switch to recaptcha v2?
> 
> @jinmvp I think @senzacionale is asking how to switch to recaptcha v2. Is that a separate configuration option? I have `provider.id` option set to `2captcha` provider, so v2 would be used automatically when running `await page.solveRecaptchas()`, is that right?

There is a button there if the site has alternative challenge. Like walmart, check the pics below. 
![image](https://user-images.githubusercontent.com/87594788/193812912-0830a170-a40d-4f46-a543-e16ade501b5f.png)
